modules/illumina.nf

```diff
@@ -74,19 +74,49 @@ process readMapping {
         """
 }
 
+process bamToCram {
+    /**
+    * Convert bam to cram using samtools with embeded reference, indexes the CRAM (http://www.htslib.org/doc/samtools.html)
+    * @input 
+    * @output 
+    */
+
+    tag { sampleName }
+
+    publishDir "${params.outdir}/${task.process.replaceAll(":","_")}", pattern: "${bam.baseName}.cram", mode: 'copy'
+    publishDir "${params.outdir}/${task.process.replaceAll(":","_")}", pattern: "${bam.baseName}.cram.crai", mode: 'copy'
+
+    input:
+        tuple(sampleName, path(bam), path(ref))
+
+    output:
+        tuple sampleName, path("${bam.baseName}.cram"), emit: cramed
+        tuple sampleName, path("${bam.baseName}.cram.crai"), emit: cramedidx
+
+    script:
+        """
+        samtools view -C -O cram,embed_ref -T ${ref} -o ${bam.baseName}.cram ${bam}
```

Comment:
If bam needs to be indexed before it can be converted to a cram, index it here, not in trimPrimerSequences.

Reply:
> Thanks. Creating the crams should be optional, so copy how, for example, the --cram option was implemented and do something similar. You can have an 'if (optionwasset) { bamToCram(...)}' in the workflow rather than a whole new workflow though.
>>  added out.nf which is imported  in nanopore and illumina workflows
> I think the bams to create the crams from should be the QC pass ones. But don't publish the crams to the qc pass dir (currently only bams are allowed), where you currently publish to should be fine.
>>  this is now take care
> The existing tests should then pass. Can you also add a test for this? You could start by copying .github/scripts/test_cram_input.sh and making changes to test that crams get created. Add the new test to .github/workflows/pull_request.yml
>> I need input cram that passes QC to test bamToCram step